Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp

```diff
@@ -67,6 +68,216 @@ ReadonlySpan<StringView> available_calendars()
+    //         i. Set calendarRec.[[Day]] to %Temporal.TimeZone.prototype.day%.
+    //     b. Else,
+    //         i. Set calendarRec.[[Day]] to ? GetMethod(calendarRec.[[Receiver]], "day").
+    //         ii. If calendarRec.[[Day]] is undefined, throw a TypeError exception.
+    // 6. Else if methodName is fields, then
+    //     a. If calendarRec.[[Receiver]] is a String, then
+    //         i. Set calendarRec.[[Fields]] to %Temporal.TimeZone.prototype.fields%.
+    //     b. Else,
+    //         i. Set calendarRec.[[Fields]] to ? GetMethod(calendarRec.[[Receiver]], "fields").
+    //         ii. If calendarRec.[[Fields]] is undefined, throw a TypeError exception.
+    // 7. Else if methodName is mergeFields, then
+    //     a. If calendarRec.[[Receiver]] is a String, then
+    //         i. Set calendarRec.[[MergeFields]] to %Temporal.TimeZone.prototype.mergeFields%.
+    //     b. Else,
+    //         i. Set calendarRec.[[MergeFields]] to ? GetMethod(calendarRec.[[Receiver]], "mergeFields").
+    //         ii. If calendarRec.[[MergeFields]] is undefined, throw a TypeError exception.
+    // 8. Else if methodName is monthDayFromFields, then
+    //     a. If calendarRec.[[Receiver]] is a String, then
+    //         i. Set calendarRec.[[MonthDayFromFields]] to %Temporal.TimeZone.prototype.monthDayFromFields%.
+    //     b. Else,
+    //         i. Set calendarRec.[[MonthDayFromFields]] to ? GetMethod(calendarRec.[[Receiver]], "monthDayFromFields").
+    //         ii. If calendarRec.[[MonthDayFromFields]] is undefined, throw a TypeError exception.
+    // 9. Else if methodName is yearMonthFromFields, then
+    //     a. If calendarRec.[[Receiver]] is a String, then
+    //         i. Set calendarRec.[[YearMonthFromFields]] to %Temporal.TimeZone.prototype.yearMonthFromFields%.
+    //     b. Else,
+    //         i. Set calendarRec.[[YearMonthFromFields]] to ? GetMethod(calendarRec.[[Receiver]], "yearMonthFromFields").
+    //         ii. If calendarRec.[[YearMonthFromFields]] is undefined, throw a TypeError exception.
+    switch (method_name) {
+#define __JS_ENUMERATE(PascalName, camelName, snake_name)                                                                                        \
+    case CalendarMethod::PascalName: {                                                                                                           \
+        VERIFY(!calendar_record.snake_name);                                                                                                     \
+        if (calendar_record.receiver.has<String>()) {                                                                                            \
+            Value prototype_value { realm.intrinsics().temporal_calendar_prototype() };                                                          \
+            calendar_record.snake_name = MUST(prototype_value.get_method(vm, vm.names.camelName));                                               \
```

Comment:
hmm I'm concerned that this is an observable deviation from the spec, as the spec does not say to use `GetMethod`. I think this should be:
```c++
calendar_record.snake_name = realm.intrinsics().temporal_calendar_prototype().get_without_side_effects(vm.names.camelName).as_function();
```

...which is quite a mouthful that can be broken up into multiple lines if you'd like. Or cache the return value of `realm.intrinsics().temporal_calendar_prototype()` outside of this macro, so we can call it like:
```c++
calendar_record.snake_name = temporal_calendar_prototype.get_without_side_effects(vm.names.camelName).as_function();
```

Reply:
Also shouldn't the prototype we access change depending on the step? Only step 1 says `%Temporal.Calendar.prototype...`. Others are `%Temporal.TimeZone.prototype...`

Reply:
Wait I'm extra confused. `Temporal.TimeZone.prototype.dateFromFields` (and the methods that follow) don't exist...? If I search for that text in the Temporal spec, it appears only in this method. Spec bug?

Reply:
Yeah, I think it must be a spec bug - maybe it was some type of copy paste error from https://tc39.es/proposal-temporal/#sec-temporal-timezonemethodsrecordlookup

Which had a similar refactor done (I also have this one sitting around locally, it's also required in the codepath that I'm trying to chase down for fixing some of the timeout issues